pkg/util/marshal/query.go

```diff
@@ -264,7 +264,7 @@ func encodeStreams(streams logqlmodel.Streams, s *jsoniter.Stream) error {
 
 		err := encodeStream(stream, s)
 		if err != nil {
-			return nil
+			return err
```

Comment:
Did a test fail in CI and that's why we're adding this? Or do we need another to test this?

Reply:
Yes, ideally we have a test case for the error case as well.

Reply:
Alas. You are right...

Reply:
...and there was a bug indeed.